oat-sa/extension-tao-itemqti
# TR-2961 text reader asset extraction

# [TR-2961](https://oat-sa.atlassian.net/browse/TR-2961)

## Summary
Need to create functionality which allow publish inline base64 encoded PCI Text Reader item assets to cdn
 
## How to test
* update extension by branch
* create `item` with `Text Reader` with image
* create `test` with created in prev step `item`
* publish this `test` as local
* run published delivery and check in browser inspector image `src`

For affecting local `env` without `cdn` need to update `taoQtiItem/model/compile/QtiAssetReplacer/NullQtiItemAssetReplacer.php`
by next code 
```
class NullQtiItemAssetReplacer extends ConfigurableService implements QtiItemAssetReplacer
{
    public function shouldBeReplaced(PackedAsset $packetAsset): bool
    {
        return $packetAsset->getMediaAsset()->getMediaSource() instanceof InlineSource;
    }

    public function replace(PackedAsset $packetAsset, $itemId): PackedAsset
    {
        $mediaAsset = $packetAsset->getMediaAsset();
        $filename = $mediaAsset->getMediaSource()->getBaseName($mediaAsset->getMediaIdentifier());

        return $packetAsset->setReplacedBy($filename);
    }
}
```
